django-backend/fecfiler/web_services/views.py

```diff
@@ -37,26 +41,38 @@ def create_dot_fec(self, request):
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
         report_id = serializer.validated_data["report_id"]
         logger.debug(f"Starting Celery Task create_dot_fec for report :{report_id}")
-        task = create_dot_fec.apply_async((report_id, None, None), retry=False)
+        file_name = f"{report_id}_{math.floor(datetime.now().timestamp())}.fec"
+        task = create_dot_fec.apply_async(
+            (report_id, None, None, False, file_name), retry=False
+        )
         logger.debug(f"Status from create_dot_fec report {report_id}: {task.status}")
-        return Response({"status": ".FEC task created"})
+        return Response({"file_name": file_name, "task_id": task.task_id})
+
+    @action(
+        detail=False,
+        methods=["get"],
+        url_path="dot-fec/check/(?P<task_id>[a-z0-9-]+)",
+    )
+    def check_dot_fec(self, request, task_id):
+        res = AsyncResult(task_id)
+        if res.ready():
+            return Response({"done": True, "id": res.get()})
+        return Response({"done": False})
 
     @action(
         detail=False,
         methods=["get"],
-        url_path="dot-fec/(?P<report_id>[a-z0-9-]+)",
+        url_path="dot-fec/(?P<id>[a-z0-9-]+)",
         renderer_classes=(DotFECRenderer,),
     )
-    def get_dot_fec(self, request, report_id):
+    def get_dot_fec(self, request, id):
         """Download the most recent .FEC created for a report
         Currently only useful for testing purposes
         """
-        committee_uuid = request.session["committee_uuid"]
-        dot_fec_record = DotFEC.objects.filter(
-            report__id=report_id, report__committee_account_id=committee_uuid
-        ).order_by("-file_name")
+
+        dot_fec_record = DotFEC.objects.filter(id=id)
```

Comment:
We still need to filter by `report__committee_account_id=committee_uuid` to limit access to your committee

Reply:
Ah that makes sense. Got that added back in.